mars/dataframe/sort/psrs.py

```diff
@@ -567,99 +506,59 @@ def execute(cls, ctx, op):
-    _n_partition = Int32Field("n_partition")
+    axis = Int32Field("axis")
+    by = ListField("by")
+    ascending = AnyField("ascending")
+    inplace = BoolField("inplace")
+    na_position = StringField("na_position")
+    n_partition = Int32Field("n_partition")
 
     # for sort_index
-    _level = ListField("level")
-    _sort_remaining = BoolField("sort_remaining")
+    level = ListField("level")
+    sort_remaining = BoolField("sort_remaining")
 
     # for shuffle reduce
-    _kind = StringField("kind")
-
-    def __init__(
-        self,
-        sort_type=None,
-        by=None,
-        axis=None,
-        ascending=None,
-        n_partition=None,
-        na_position=None,
-        inplace=None,
-        kind=None,
-        level=None,
-        sort_remaining=None,
-        output_types=None,
-        **kw
-    ):
-        super().__init__(
-            _sort_type=sort_type,
-            _by=by,
-            _axis=axis,
-            _ascending=ascending,
-            _n_partition=n_partition,
-            _na_position=na_position,
-            _inplace=inplace,
-            _kind=kind,
-            _level=level,
-            _sort_remaining=sort_remaining,
-            _output_types=output_types,
-            **kw
-        )
-
-    @property
-    def sort_type(self):
-        return self._sort_type
+    kind = StringField("kind")
 
-    @property
-    def by(self):
-        return self._by
-
-    @property
-    def axis(self):
-        return self._axis
-
-    @property
-    def ascending(self):
-        return self._ascending
-
-    @property
-    def inplace(self):
-        return self._inplace
-
-    @property
-    def na_position(self):
-        return self._na_position
-
-    @property
-    def level(self):
-        return self._level
-
-    @property
-    def sort_remaining(self):
-        return self._sort_remaining
-
-    @property
-    def n_partition(self):
-        return self._n_partition
-
-    @property
-    def kind(self):
-        return self._kind
+    def __init__(self, output_types=None, **kw):
+        super().__init__(_output_types=output_types, **kw)
 
     @property
     def output_limit(self):
         return 1
 
     @staticmethod
     def _calc_poses(src_cols, pivots, ascending=True):
+        class ReversedValue:
+            def __init__(self, value):
+                self._value = value
+
+            def __lt__(self, other):
+                if isinstance(other, ReversedValue):
+                    # may happen when call searchsorted
+                    return self._value >= other._value
+                return self._value >= other
+
+            def __gt__(self, other):
+                return self._value <= other
+
+            def __repr__(self):
+                return repr(self._value)
+
+        if isinstance(ascending, list):
+            for asc, col in zip(ascending, pivots.columns):
+                if not asc:
+                    if np.issubdtype(pivots.dtypes[col], np.number):
```

Comment:
Please use `pandas.api.types.is_numeric_dtype` instead, refer to https://pandas.pydata.org/docs/reference/api/pandas.api.types.is_numeric_dtype.html . numpy dtype is not compatible with pandas extension type.